service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerDateOfIssueMidEventTest.java

```diff
@@ -0,0 +1,208 @@
+
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = caseData.getJudgeAndLegalAdvisor();
+
+        assertThat(judgeAndLegalAdvisor.getAllocatedJudgeLabel()).isEqualTo(
+            "Case assigned to: His Honour Judge Richards");
+        assertThat(judgeAndLegalAdvisor.getJudgeTitle()).isEqualTo(HIS_HONOUR_JUDGE);
+        assertThat(judgeAndLegalAdvisor.getJudgeLastName()).isEqualTo("Davidson");
+    }
+
+    @Test
+    void shouldSetAssignJudgeLabelWhenAllocatedJudgeIsPopulated() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("allocatedJudge", buildAllocatedJudge()))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = caseData.getJudgeAndLegalAdvisor();
+
+        assertThat(judgeAndLegalAdvisor.getAllocatedJudgeLabel())
+            .isEqualTo("Case assigned to: His Honour Judge Richards");
+    }
+
+    @Test
+    void shouldPopulateUseAllocatedJudgeWithYesWhenJudgeAndAllocatedJudgeAreEqual() {
+        CaseDetails caseDetails = buildSameJudgeCaseDetails(buildAllocatedJudge());
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = caseData.getJudgeAndLegalAdvisor();
+
+        assertThat(judgeAndLegalAdvisor.getUseAllocatedJudge()).isEqualTo("Yes");
+    }
+
+    @Test
+    void shouldNotSetAssignedJudgeLabelIfAllocatedJudgeNotSet() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("judgeAndLegalAdvisor", JudgeAndLegalAdvisor.builder().build()))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = caseData.getJudgeAndLegalAdvisor();
+
+        assertThat(judgeAndLegalAdvisor.getAllocatedJudgeLabel()).isNull();
+    }
+
+    private CaseDetails buildSameJudgeCaseDetails(Judge judge) {
+        return CaseDetails.builder()
+            .data(Map.of(
+                "allocatedJudge", judge,
+                "standardDirectionOrder", Order.builder()
+                    .directions(buildDirections(createDirections()))
+                    .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                        .judgeTitle(judge.getJudgeTitle())
+                        .judgeLastName(judge.getJudgeLastName())
+                        .build())
+                    .build()))
+            .build();
+    }
+
+    private Judge buildAllocatedJudge() {
```

Comment:
we have a testJudge method in testDataHelper we could probably use now instead of this - don't mind if you want to leave as is as would require changing the asserts for some of the tests

Reply:
I will update to use that, thanks